mobile/config.go

```diff
@@ -19,21 +21,31 @@ import (
 )
 
 type Config struct {
-	Identity  []byte
+	// IdentitySeed is the seed that is used to generate Ed25519 libp2p key-pair. If unspecified, a
+	// random seed is used, otherwise, it must be 32 bytes long. Longer seed value is accepted but
+	// only the first 32 bytes are read. See: ed25519.SeedSize.
+	IdentitySeed []byte
+	// StorePath is the path used to persist data. If unspecified, an in-memory store is used.
 	StorePath string
 }
 
 func (cfg *Config) init(mc *Client) error {
-	var err error
-	var hopts []libp2p.Option
-	if len(cfg.Identity) != 0 {
-		pk, err := crypto.UnmarshalPrivateKey(cfg.Identity)
-		if err != nil {
-			return err
-		}
-		hopts = append(hopts, libp2p.Identity(pk))
+	seedLen := len(cfg.IdentitySeed)
+	var seed io.Reader
+	switch {
+	case seedLen == 0:
+		// No seed is given; generate identity at random by leaving seed to be nil.
+		break
+	case seedLen < ed25519.SeedSize:
+		return fmt.Errorf("identity seed too short; it must be at least 32 bytes but got %d", seedLen)
+	default:
+		seed = bytes.NewBuffer(cfg.IdentitySeed)
+	}
+	pk, _, err := crypto.GenerateEd25519Key(seed)
```

Comment:
Hi Masih, had a talk with @Ruffiano and it seems it makes more sense to send the public/private keys from javascript to go instead of hash and I think the crypto library supports that option as well:
func GenerateKey(rand [io](https://pkg.go.dev/io).[Reader](https://pkg.go.dev/io#Reader)) ([PublicKey](https://pkg.go.dev/crypto/ed25519#PublicKey), [PrivateKey](https://pkg.go.dev/crypto/ed25519#PrivateKey), [error](https://pkg.go.dev/builtin#error))

Can we change the seed to public/private key?
The reason is we need the same keypair on both javascript and go, but right now we are creating it from hash on javascript and separately creating it from the hash on go. Theoretically, they should end up being the same, but it doesn't look clean to me to generate it in two places if they should be the same

Reply:
Hi @masih the only thing needed on javascript side is to be bale to get the generated libp2p PeerId, since on javascript the function is not available due to library limitations

Reply:
Thank you. closing this PR and will add in a separate PR